src/Validators/Validators.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace App\Validators;
+
+use App\Interfaces\ValidationInterface;
+use Exception;
+
+class Validators implements ValidationInterface
+{
+
+    public static function ValidateNewPost($unvalidated): array
+    {
+        $unvalidated->rule('required', ['title', 'author', 'date', 'post']);
+        $unvalidated->rule('lengthMin', 'title', 1);
+        $unvalidated->rule('lengthMin', 'author', 1);
+        $unvalidated->rule('dateFormat', 'date', 'd-m-Y');
+        $unvalidated->rule('lengthMin', 'post', 5);
+        if ($unvalidated->validate()) {
+            return $unvalidated;
+        } else {
+            throw new Exception('Please ensure all required fields are filled and meet requirements.');

```

Comment:
don't really need the else bit here, also i think you need to take a look at how you want to handle errors. you are throwing an exception here but aren't catching it anywhere where you are calling this method